x-pack/plugins/siem/public/common/components/exceptions/helpers.tsx

```diff
@@ -0,0 +1,198 @@
+  Comment,
+  ExceptionListItemSchema,
+} from './types';
+import { EXCEPTION_OPERATORS, isOperator } from './operators';
+
+export const getEmptyValue = () => '—';
+
+/**
+ * Returns the operator type, may not need this if using io-ts types
+ *
+ * @param entry a single ExceptionItem entry
+ */
+export const getOperatorType = (entry: ExceptionEntry): OperatorType => {
+  switch (entry.type) {
+    case 'nested':
+    case 'match':
+      return OperatorType.PHRASE;
+    case 'match_any':
+      return OperatorType.PHRASES;
+    case 'list':
+      return OperatorType.LIST;
+    default:
+      return OperatorType.EXISTS;
+  }
+};
+
+/**
+ * Determines operator selection (is/is not/is one of, etc.)
+ * Default operator is "is"
+ *
+ * @param entry a single ExceptionItem entry
+ */
+export const getExceptionOperatorSelect = (entry: ExceptionEntry): OperatorOption => {
+  const operatorType = getOperatorType(entry);
+  const foundOperator = EXCEPTION_OPERATORS.find((operatorOption) => {
+    return entry.operator === operatorOption.operator && operatorType === operatorOption.type;
+  });
+
+  return foundOperator ?? isOperator;
+};
+
+export const determineIfIsNested = (
+  tbd: ExceptionEntry | NestedExceptionEntry
+): tbd is NestedExceptionEntry => {
+  if (tbd.type === 'nested') {
+    return true;
+  }
+  return false;
+};
+
+/**
+ * Formats ExceptionItem entries into simple field, operator, value
+ * for use in rendering items in table
+ *
+ * @param entries an ExceptionItem's entries
+ */
+export const getFormattedEntries = (
+  entries: Array<ExceptionEntry | NestedExceptionEntry>
+): FormattedEntry[] => {
+  const formattedEntries = entries.map((entry) => {
+    if (determineIfIsNested(entry)) {
+      const parent = { fieldName: entry.field, operator: null, value: null, isNested: false };
+      return entry.entries.reduce<FormattedEntry[]>(
+        (acc, nestedEntry) => {
+          const formattedEntry = formatEntry({
+            isNested: true,
+            parent: entry.field,
+            item: nestedEntry,
+          });
+          return [...acc, { ...formattedEntry }];
+        },
+        [parent]
+      );
+    } else {
+      return formatEntry({ isNested: false, item: entry });
+    }
+  });
+
+  return flatten(formattedEntries);
+};
+
+/**
+ * Helper method for `getFormattedEntries`
+ */
+export const formatEntry = ({
+  isNested,
+  parent,
+  item,
+}: {
+  isNested: boolean;
+  parent?: string;
+  item: ExceptionEntry;
+}) => {
+  const operator = getExceptionOperatorSelect(item);
+  const operatorType = getOperatorType(item);
+  const value = operatorType === OperatorType.EXISTS ? null : item.value;
+
+  return {
+    fieldName: isNested ? `${parent}.${item.field}` : item.field,
+    operator: operator.message,
+    value,
+    isNested,
+  };
+};
+
+export const getOperatingSystems = (tags: string[]): string | null => {
+  const [includesOs, osMatch] = getTagsInclude({ tags, regex: /(?<=\b)(os:[^;]*)(?=\b)/ });
```

Comment:
You can leave this as is it is tricky to read the regex, but if I am reading it correctly and you're looking for "os:" at the beginning can't it be just this:

```ts
['os:linux', 'os:windows', 'process:notepad.exe'].filter(tag => tag.startsWith('os:')).map(os => os.substring(3).toUpperCase())
["LINUX", "WINDOWS"]
```

You avoid the lodash capitalize and if you need to add a trim in the chain you can as well.

Just a suggestion, you can keep the regex if you like it more.

Reply:
So initially I was trying to be super flexible and have it work for if a user specified an os one at a time `os:mac` or comma separated `os:mac, windows, etc`. But this is for out internal `_tags` so I think we can maintain a format of specifying os in separate strings. I updated to use your suggested code, but kept the use of `capitalize` as the UI calls for just capitalizing the first letter, not the whole word. 

Actually wonder if we have a util for that? I'll take a look.